utils/generate_profile.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+
+# Get input file
+# Determine format (CSV or something else like OSCAL)
+# Parse input into a data structure (dictionary)
+
+import abc
+import argparse
+import os
+import re
+
+import json
+import yaml
+
+import pandas
+
+DESCRIPTION = '''
+A tool for converting benchmarks to profiles.
+'''
+
+
+class LiteralUnicode(str):
+    pass
+
+
+def literal_unicode_representer(dumper, data):
+    return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='|')
+
+
+yaml.add_representer(LiteralUnicode, literal_unicode_representer)
+
+
+def setup_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description=DESCRIPTION)
+    parser.add_argument('-i', '--input-file', required=True)
+    parser.add_argument('-p', '--product-type', required=True)
+
+    subparsers = parser.add_subparsers()
+    list_parser = subparsers.add_parser('list', help='List controls within a benchmark')
+    list_parser.set_defaults(func=list_controls)
+    generate_parser = subparsers.add_parser('generate', help='Generate a control from benchmark')
+    generate_parser.add_argument('-c', '--control', required=True, help='Control ID to generate')
+    generate_parser.set_defaults(func=generate_control)
+
+    return parser
+
+
+class Parser(abc.ABC):
+
+    @abc.abstractmethod
+    def __init__(self) -> None:
+        raise NotImplemented
+
+    @abc.abstractmethod
+    def get_name(self):
+        raise NotImplemented
+
+    @abc.abstractmethod
+    def get_version(self):
+        raise NotImplemented
+
+    @abc.abstractmethod
+    def parse(self):
+        raise NotImplemented
+
+
+class Section:
+    def __init__(self, id, title):
+        self.id = id
+        self.title = title
+        self.description = None
+        self.sections = []
+        self.controls = []
+
+
+class Control:
+    def __init__(self, id, title):
+        self.id = id
+        self.title = title
+        self.description = None
+        self.level = None
+        self.remediation = None
+        self.rationale = None
```

Comment:
to bring a slack conversation over here and tag @Mab879 - would it be a good idea to merge this class with the `control` class from `srg/controls.py`?

Reply:
There is another version of this in https://github.com/ComplianceAsCode/content/pull/10469 

@Mab879 what are your thoughts on a shared library?

Reply:
I think that shared library would sense.

Reply:
Ok - I'll help get the deps squared away in https://github.com/ComplianceAsCode/content/pull/10474 and https://github.com/ComplianceAsCode/content/pull/10473. Then I'll start pulling the common bits into a shared library that we can import as a python-specific requirement.

Reply:
I threw together a quick library https://github.com/rhmdnd/pycompliance